### ISSUE TITLE
Indentation is wrong for LABEL:'s inside a block

### ISSUE THREAD

**Issue by akarelas** [Camelcade/Perl5-IDEA#2320]
IDE: WebStorm 2020.2.3 #WS-202.7660.23
Perl5 plugin: 2020.2.4
OS: Ubuntu MATE 20.04

When I type this:

```perl
{
    FOO: {
```

and press Enter, I get this, that has very wrong indentation:

```perl
{
    FOO: {
    |                        # cursor is at: |
}
}
```

...and the cursor is right underneath F (of FOO).

**Comment by hurricup** [comment]
Looks like there is a bug in the formatting engine in IDEA. Investigating. 
As temporary workaround you may add a newline between label and block.

**Comment by hurricup** [comment]
I did. But it's complicated.... 
Looks like perl formatter has some bug, which usually works as expected because of other bug in platform. But this case fails :)

**Comment by akarelas** [comment]
So is it a matter of finding this Camelcade bug? Or does the other bug in IDEA need to be found and solved too?

**Comment by hurricup** [comment]
yes. basically first perl formatter should be refactored to work properly

**Comment by akarelas** [comment]
If someone wanted to help you in debugging this, would they need to know Java?

**Comment by hurricup** [comment]
Yes and it's just for start.
There is no need to debug actually, it requires a time to work I don't currently have.
